db/index_changes.go

```diff
@@ -403,17 +422,105 @@ func (db *Database) initializeChannelFeeds(channelsSince channels.TimedSet, opti
 	return feeds, nil
 }
 
+// Calculates the range for backfill processing, for compatibility with changes streaming.
+//  For the given:
+//    - since clock SinceClock
+//    - current backfill position vb-B.seq-B,
+//    - granting sequence vb-G.seq-G
+//  We want to return everything for the channel from vb-B.seq-B up to vb-G.seq-G as backfill,
+//  and also return everything from vb-G.seq-G to vb-max.seq-max that's earlier the since value,
+//
+// From Clock:
+// If vb < vb-B, seq = MaxUint
+// If vb = vb-B, seq = seq-b
+// if vb > vb-B, seq = SinceClock seq
+// To Clock:
+// If vb < vb-G, seq = MaxInt
+// If vb = vb-G, seq = seq-G
+// if vb > vb-G, seq = SinceClock
+
+func calculateBackfillRange(backfillPosition base.VbSequence, triggerPosition base.VbSequence, sinceClock base.SequenceClock) (fromClock, toClock base.SequenceClock) {
+
+	fromClock = base.NewSequenceClockImpl()
+	toClock = base.NewSequenceClockImpl()
+	MAX_SEQUENCE := uint64(math.MaxUint64)
+
+	for vbInt, _ := range sinceClock.Value() {
+		vbNo := uint16(vbInt)
+		var fromSeq, toSeq uint64
+
+		// Calculate from sequence for vbucket
+		if vbNo < backfillPosition.Vb {
+			fromSeq = MAX_SEQUENCE
+		} else if vbNo > backfillPosition.Vb {
+			fromSeq = sinceClock.GetSequence(vbNo)
+		} else { // vbNo == backfillPosition.Vb
```

Comment:
The panic seems redundant for code that should be impossible to hit (greater than, less than, equals), which was sort of the idea behind the original code just falling into an 'else' for the last possible case.  

I agree about the readability, though - it's probably better to have 'else ==' than else // ==, even if it's redundant.